packages/runtime/container-runtime/src/test/gc/garbageCollection.spec.ts

```diff
@@ -1672,267 +1676,98 @@ describe("Garbage Collection Tests", () => {
-				clock.tick(defaultSweepTimeoutMs + 1);
-
-				// There should be 2 deleted nodes and data stores. There shouldn't be any nodes whose
-				// reference state updated.
-				expectedStats.deletedNodeCount = 2;
-				expectedStats.deletedDataStoreCount = 2;
-				expectedStats.updatedNodeCount = 0;
-				expectedStats.updatedDataStoreCount = 0;
-
-				gcStats = await garbageCollector.collectGarbage({});
-				assert.deepStrictEqual(gcStats, expectedStats, "Incorrect GC stats 1");
-
-				// Unreference another data store node.
-				defaultGCData.gcNodes[nodes[0]] = [];
-
-				if (sweepEnabled) {
-					// If sweep is enabled, there should be 2 less nodes / data stores since they got deleted.
-					// There should be 1 new unreferenced node / data store.
-					expectedStats.nodeCount -= 2;
-					expectedStats.dataStoreCount -= 2;
-					expectedStats.unrefNodeCount = 1;
-					expectedStats.unrefDataStoreCount = 1;
-				} else {
-					// If sweep is disabled, there should be 1 more unreferenced node / data store.
-					expectedStats.unrefNodeCount++;
-					expectedStats.unrefDataStoreCount++;
-				}
-				// There should be 1 node / data store whose reference state got updated.
-				expectedStats.updatedNodeCount = 1;
-				expectedStats.updatedDataStoreCount = 1;
-
-				gcStats = await garbageCollector.collectGarbage({});
-				assert.deepStrictEqual(gcStats, expectedStats, "Incorrect GC stats 2");
-
-				// Advance the clock past sweep timeout again so that unreferenced node is deleted.
-				clock.tick(defaultSweepTimeoutMs + 1);
-
-				// No nodes are updated since the last run.
-				// There should be 1 more deleted node / data store.
-				expectedStats.updatedNodeCount = 0;
-				expectedStats.updatedDataStoreCount = 0;
-				expectedStats.deletedNodeCount++;
-				expectedStats.deletedDataStoreCount++;
-
-				gcStats = await garbageCollector.collectGarbage({});
-				assert.deepStrictEqual(gcStats, expectedStats, "Incorrect GC stats 3");
-
-				if (sweepEnabled) {
-					// If sweep is enabled, there should be 1 less node / data store since it got deleted.
-					// There shouldn't be any unreferenced node / data store.
-					expectedStats.nodeCount--;
-					expectedStats.dataStoreCount--;
-					expectedStats.unrefNodeCount = 0;
-					expectedStats.unrefDataStoreCount = 0;
-				}
+		beforeEach(async () => {
+			garbageCollector = createGarbageCollector({ gcOptions: { gcSweepGeneration: 1 } });
+		});
 
-				gcStats = await garbageCollector.collectGarbage({});
-				assert.deepStrictEqual(gcStats, expectedStats, "Incorrect GC stats 4");
-			});
+		it("can submit GC op compat behavior", async () => {
+			type GarbageCollectorWithSubmit = Omit<GarbageCollector, "submitMessage"> & {
+				submitMessage(message: ContainerRuntimeGCMessage): void;
+			};
+			const gcWithSubmit = garbageCollector as GarbageCollectorWithSubmit;
+			const gcMessage: Omit<GarbageCollectionMessage, "type" | "deletedNodeIds"> & {
```

Comment:
`Omit<GarbageCollectionMessage, "type" | "deletedNodeIds">` is kind of strange to me, it's an empty object.

Reply:
Changed it to Record<string, unknown>